target/riscv/translate.c

```diff
@@ -1173,3 +1173,8 @@ void riscv_translate_init(void)
     load_val = tcg_global_mem_new(cpu_env, offsetof(CPURISCVState, load_val),
                              "load_val");
 }
+
+void gen_cheri_break_loadlink(TCGv_cap_checked_ptr out_addr, TCGv_i32 memop)
+{
+    tcg_gen_movi_tl((TCGv)load_res, -1);
```

Comment:
Seems like this line is causing a fatal error during RISC-V QEMU boot.
```
qemu-system-riscv64cheri: FATAL: cheri_tag_invalidate: ffffffffffffffff+8 crosses a page boundary
qemu-system-riscv64cheri: Probably caused by guest instruction: 0xffffffc000278736:  18a4b62f          sc.d            a2,a0,(s1)
```

It seems like the RISC-V SC implementation uses load_res directly (rather than the argument to sc) to avoid needing a local TCGTemp (since it needs to survive past the addr==load_res check):
```
    tcg_gen_atomic_cmpxchg_tl_with_checked_addr(src1, load_res, load_val, src2, ctx->mem_idx, mop);
```

RISC-V regression tests running for each pull request would be nice 😢 .
I'll comment this out for now and will look into a fix tomorrow.